File: package/example/android/app/src/main/java/com/amazingsoftworks/filesystem_picker_example/FileSystemPickerPlugin.java
package com.amazingsoftworks.filesystem_picker_example;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.HashMap;

import io.flutter.embedding.android.FlutterActivity;
import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugins.GeneratedPluginRegistrant;

import static android.os.Environment.isExternalStorageManager;

public class FileSystemPickerPlugin extends FlutterActivity {

  private static final String channel = "filesystem_picker/android_native";

  @Override
  public void configureFlutterEngine(@NonNull FlutterEngine flutterEngine) {
    GeneratedPluginRegistrant.registerWith(flutterEngine);
    new MethodChannel(flutterEngine.getDartExecutor().getBinaryMessenger(), channel)
            .setMethodCallHandler(
                    (call, result) -> {
                      if (call.method.equals("isExternalStorageManager")) {
                          if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.R) {
                              result.success(isExternalStorageManager());
                          } else {
                              result.success(true);
                          }
                      } else if (call.method.equals("getExtStorageData")) {
                          ArrayList<HashMap<String, Object>> reply = StorageUtils.getExternalStorageAvailableData(getApplicationContext());
                          result.success(reply);
                      } else {
                        result.notImplemented();
                      }
                    }
            );
  }
}

File: package/android/src/main/kotlin/com/amazingsoftworks/filesystem_picker_plugin/FilesystemPickerPlugin.kt
package com.amazingsoftworks.filesystem_picker_plugin

import android.content.Context
import android.os.Build
import android.os.Environment
import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** FilesystemPickerPlugin */
class FilesystemPickerPlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel
  private lateinit var context: Context

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    context = flutterPluginBinding.applicationContext
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "filesystem_picker/android_native")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    if (call.method == "getPlatformVersion") {
      result.success("Android ${Build.VERSION.RELEASE}")
    } else if (call.method == "isExternalStorageManager") {
      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
        result.success(Environment.isExternalStorageManager())
      } else {
        result.success(true)
      }
    } else if (call.method == "getExtStorageData") {
      val reply = StorageUtils.getExternalStorageAvailableData(context)
      result.success(reply)
    } else {
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
